src/Apps/Settings/Routes/EditProfile/Components/SettingsEditProfileFields.tsx

```diff
@@ -0,0 +1,221 @@
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import { editProfileVerificationSchema } from "Apps/Auction/Components/Form/Utils"
+import { CollectorProfileHeaderAvatarFragmentContainer } from "Apps/CollectorProfile/Components/CollectorProfileHeader/Components/CollectorProfileHeaderAvatar"
+import {
+  LocationAutocompleteInput,
+  normalizePlace,
+} from "Components/LocationAutocompleteInput"
+import { useFormik } from "formik"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useUpdateMyUserProfile } from "Utils/Hooks/Mutations/useUpdateMyUserProfile"
+import { SettingsEditProfileFields_me$data } from "__generated__/SettingsEditProfileFields_me.graphql"
+import { EditableLocation } from "__generated__/useUpdateMyUserProfileMutation.graphql"
+
+interface EditableLocationProps extends EditableLocation {
+  display: string | null
+}
+
+interface EditProfileFormModel {
+  photo: string
+  name: string
+  displayLocation: { display: string | null }
+  location: EditableLocationProps | null
+  profession: string
+  otherRelevantPositions: string
+  bio: string
+}
+
+interface SettingsEditProfileFieldsProps {
+  me: SettingsEditProfileFields_me$data
+}
+
+const SettingsEditProfileFields: React.FC<SettingsEditProfileFieldsProps> = ({
+  me,
+}) => {
+  const { submitUpdateMyUserProfile } = useUpdateMyUserProfile()
+  const { values, isSubmitting, handleSubmit, setFieldValue } = useFormik<
+    EditProfileFormModel
+  >({
+    initialValues: {
+      name: me.name ?? "",
+      displayLocation: { display: me.location?.display ?? null },
+      location: me.location ?? null,
+      profession: me.profession ?? "",
+      otherRelevantPositions: me.otherRelevantPositions ?? "",
+      bio: me.bio ?? "",
+      photo: me.icon?.url ?? "",
+    },
+    onSubmit: async values => {
+      try {
+        const newLocation = { ...values.location }
+        delete newLocation.display
+        const payload = {
+          name: values.name,
+          location: newLocation,
+          profession: values.profession,
+          otherRelevantPositions: values.otherRelevantPositions,
+          bio: values.bio,
+        }
+
+        await submitUpdateMyUserProfile(payload)
+      } catch (error) {
+        console.error("Failed to update user profile ", error)
+      }
+    },
+    validationSchema: editProfileVerificationSchema,
+    validateOnBlur: true,
+  })
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <Join separator={<Spacer y={4} />}>
+          <Flex alignItems="center">
+            <CollectorProfileHeaderAvatarFragmentContainer me={me} />
+
+            <Text>Choose an Image</Text>
+          </Flex>
+
+          <Input
+            title="Full name"
+            placeholder="Full name"
+            name="full-name"
+            required
+            maxLength={256}
+            value={values.name}
+            onChange={({ target: { value } }) => {
+              setFieldValue("name", value)
+            }}
```

Comment:
**suggestion:** You could use Formik' [handleChange](https://formik.org/docs/api/formik#handlechange-e-reactchangeeventany--void) helper here ([example](https://github.com/artsy/force/blob/main/src/Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/Components/ArtworkDetailsForm.tsx#L235)).

```suggestion
            onChange={handleChange}
```

Reply:
Hmmm can't make this change quickly
@MrSltun did you implement it like that for a reason?

Reply:
Hmm, shouldn't replacing `setFieldValue` with `onChange` just work for most of these inputs?

Reply:
In this case I can't edit the initial value

Reply:
I was testing `handleChange` but it wasn't updating the values for some reason, so I replaced it with the current implementation

Reply:
maybe `handleChange` didn't work since the field name is different than the value name? (like `name="full-name"` while the value is `name`)

Reply:
Oh wow, looks like you're right

Reply:
Works with everything, but breaks the location field
I'd keep it as it is, location requires custom updates, with normalize function

Reply:
Let's change it to `handleChange` wherever it is possible and use a custom handler for the location field ([similar to the SWA form](https://github.com/artsy/force/blob/main/src/Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/Components/ArtworkDetailsForm.tsx#L192)).

Reply:
I had an [issue with the `<LocationAutocompleteInput />` component again](https://artsy.slack.com/archives/CP9P4KR35/p1662033427441369), didn't realise that yesterday
Now all is working with the suggested changes  🎉